### PR TITLE
Refactor to use components and support IPC

### LINKED ISSUE #24: Convert TelloDriver to a component and support IPC
Convert TelloDriver to a ROS2 component. Set use_intra_process_comms = true.

### PR DESCRIPTION
**Pull request by clydemcqueen**
An early look, would like feedback on naming and structure
A fairly big change, in part due to auto-format whitespace changes that I missed before
Fixes #24 
Tested w/ emulator
Needs more testing, and I'd like take another look at deps to see if we can drop cv_bridge

### CODE REVIEW

**Comment by ptrmu** [comment]
I think the name changes are really helpful. It is way more obvious which is a node.

**Comment by ptrmu** [comment]
Dropping cv_bridge would be nice. But maybe it is useful to ensure compatibility between ROS2 and OpenCV image formats in the future. On the other hand neither of those systems is going to change its image format within the lifetime of our code. :)

**Comment by clydemcqueen** [comment]
Looking at the code, it should be easy to get rid of both cv_bridge and OpenCV deps. In fact, there's an issue to track this: https://github.com/clydemcqueen/tello_ros/issues/6

I'm going to leave this as-is for now.

**Comment by clydemcqueen** [comment]
Tested w/ a real drone.
Tested w/ `ros2 component xxx` commands.

Not yet done: testing manual composition and IPC between tello_driver and fiducial_vlam. I think this test will be interesting, but I won't check in the resulting code, since I don't really want to add a dep on fiducial_vlam.

File: tello_driver/CMakeLists.txt [hunk 1]
```diff
@@ -99,24 +107,72 @@ target_compile_definitions(tello_emulator
...
+
+#=============
+# Export incantations, see https://github.com/ros2/demos/blob/master/composition/CMakeLists.txt
+#=============
+
+if (NOT WIN32)
+  ament_environment_hooks("${ament_cmake_package_templates_ENVIRONMENT_HOOK_LIBRARY_PATH}")
+endif ()
+
+#=============
```
Comment (ptrmu) [inline tello_driver/CMakeLists.txt]: So at this point, we cannot use Clion to debug the code in tello_driver?

File: tello_driver/CMakeLists.txt [hunk 2]
```diff
@@ -29,44 +29,49 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
 # Find packages
 find_package(ament_cmake REQUIRED)
 find_package(camera_calibration_parsers REQUIRED)
+find_package(class_loader REQUIRED)
```
Comment (ptrmu) [inline tello_driver/CMakeLists.txt]: Is class_loader needed if we are not doing launch file composition? Is there any reason to add the RCLCPP_COMPONENTS_REGISTER_NODE macro if we are not doing launch file composition?

Comment (clydemcqueen) [reply tello_driver/CMakeLists.txt]: Correct, class_loader and RCLCPP_COMPONENTS_REGISTER_NODE macro are only required for launch file and CLI (`ros2 component load ...`) composition.

File: tello_driver/CMakeLists.txt (hunk 1 above)
Comment (clydemcqueen) [reply tello_driver/CMakeLists.txt]: Ah, right. I'll add a _debug executable.

Comment (clydemcqueen) [reply tello_driver/CMakeLists.txt]: PTAL, the _main executables can be debugged in CLion.

File: tello_driver/CMakeLists.txt [hunk 3]
```diff
@@ -29,44 +29,46 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
...
+# Package includes not needed for CMake >= 2.8.11
 include_directories(
   include
   h264decoder
-  ${camera_calibration_parsers_INCLUDE_DIRS}
-  ${geometry_msgs_INCLUDE_DIRS}
   ${ros2_shared_INCLUDE_DIRS}
-  ${sensor_msgs_INCLUDE_DIRS}
-  ${std_msgs_INCLUDE_DIRS}
   ${tello_msgs_INCLUDE_DIRS})
```
Comment (ptrmu) [inline tello_driver/CMakeLists.txt]: doesn't find_package take care of this for ros2_shared and tello_msgs?

Comment (ptrmu) [reply tello_driver/CMakeLists.txt]: Maybe these lines cause dependencies to be generated and we want those dependencies for our projects that might change.

Comment (clydemcqueen) [reply tello_driver/CMakeLists.txt]: AFAIK these lines do not change the dependencies. Colcon runs fine when I remove these lines. In fact, so does CLion. So, I removed them.

IIRC CLion can't find the include path when you first add a dependency on a package that is built in the Colcon workspace. Adding a line like this makes it happy. But at some point CLion gets happy again -- maybe after a full Colcon build? -- and we can remove the include. Next time this happens I'll try to take the time to investigate.

**Pull request merged**